src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -109,12 +103,8 @@ public ReadOnlyTensorSpan(T[]? array, int start, scoped ReadOnlySpan<nint> lengt
                     ThrowHelper.ThrowArgument_InvalidStridesAndLengths();
             }
 
-            _flattenedLength = linearLength;
-            _memoryLength = array.Length;
+            _shape = new TensorShape(array.Length - start, lengths, strides);
```

Comment:
nit: I wonder if some of the validation logic could be in a helper on `TensorShape`

-- Not needed to fix in this PR, just a possible simplification opportunity